### PR TITLE
190 shop bugs

### LINKED ISSUE #190: Fix Shop Bugs
- [x] Default soundtrack crash
- [ ] Visibility of buttons
- [ ] Preview buttons

### PR DESCRIPTION
**Pull request by MKerkhof**
Will close #190 

Changed the visibility of the shop. The shop now updates everytime you buy something, so does the visibility of the buttons. Also fixed the faulty path of the radioactive track.

<user-interface, bug

### CODE REVIEW

**Comment by mpsijm** [comment]
Thanks for fixing these bugs, they're looking good! :) I also don't have anything to comment about your code, ready to be merged once Travis is happy!

<review, user-interface

**Comment by MathiasMeuleman** [comment]
Please merge master, then this is ready to go :)

<review

**Comment by mdingjan** [comment]
As we already discussed last tuesday, please have a look at the preview buttons, as they don't work yet. In my opinion you can do that in this branch, a new issue is not really needed.
And indeed, please merge the master into this branch. :)

<review,feature

**Comment by MKerkhof** [comment]
Implemented the preview buttons and master is merged! Couldn't find any SAT warnings left so please review again :)

<review, feature

**Comment by mpsijm** [comment]
The preview is finally there and working correctly :D once you merged master and Travis is happy, this branch can be merged :)

<review, feature, user-interface

**Comment by thervh70** [comment]
There is still a bug left: if i have a total of 100 coins i cannot buy an item that costs 100 coins.

< bug

**Comment by mdingjan** [comment]
I've found one more bug:
Disable the soundtrack in SettingsScene -> Play a random soundtrack preview -> go to the main menu and click on 'Play' (the preview is still playing) -> in GameScene the soundtrack of the game will be activated and won't stop. 
Try it out yourself. ;)

<bug

File: src/main/java/nl/tudelft/ti2206/group9/gui/popup/AbstractInfoPopup.java [hunk 1]
```diff
@@ -97,7 +97,9 @@ public Button getRightButton() {
         return right;
     }
 
-    /** Plays the button sound once, given a volumelevel. */
+    /**
+     * Plays the button sound once, given a volumelevel.
+     */
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/gui/popup/AbstractInfoPopup.java]: Besides, why did you change so many javadoc to the style above? I guess this has been autogenerated and it has been done for a lot of classes. :P 

<documentation

Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/popup/AbstractInfoPopup.java]: Yeah, I guessed Maikel can't control his mighty IntelliJ XD I already reverted the most annoying ones (and the checkstyle-warning creating ones) ;p

<review, documentation, layout

**Comment by thervh70** [comment]
The problem stated by @mdingjan can be solved by making the preview obey the settings set by the user. For example: the preview should have the same volume level as the settings volume slider.

< bug

**Comment by MKerkhof** [comment]
Fixed all your bugs, think it's ready now :)

<bug, review

**Comment by mpsijm** [comment]
Can't click exit when shop is playing? Is that useful?
I would prefer the preview to stop when clicking back.

<review, user-interface

**Comment by mdingjan** [comment]
Thanks for processing my feedback. I just fixed the last PMD warnings.

Ready for merging!

<review

**Comment by mpsijm** [comment]
Okay, the last bugs seem to be fixed :) @thervh70 is fixing the last merge conflicts due to prices, after that this branch can be merged :D Oh wait I see merge is already complete :P @mdingjan, on your merge command we go! 

<review, feature, bug

**Comment by mdingjan** [comment]
I've checked for bugs once again and indeed, the bug has been fixed. Good job!
Will merge!

<feature,review

**Pull request merged**